api/v1/verticadb_webhook.go

```diff
@@ -1170,12 +1164,23 @@ func (v *VerticaDB) validateSubclustersInSandboxes(allErrs field.ErrorList) fiel
 
 	// check if a non-existing subcluster is defined in a sandbox
 	scMap := v.GenSubclusterMap()
+	// Also keep track of the number of sandbox primary in a sandbox. We can
+	// only have at most 1.
+	primarySubclustersInSandboxes := map[string]int{}
 	for sc, i := range seenScWithSbIndex {
 		if scInfo, ok := scMap[sc]; !ok {
 			err := field.Invalid(path.Index(i),
 				sandboxes[i],
 				fmt.Sprintf("subcluster %s does not exist", sc))
 			allErrs = append(allErrs, err)
+		} else if scInfo.IsSandboxPrimary() {
+			primarySubclustersInSandboxes[sandboxes[i].Name]++
+			if primarySubclustersInSandboxes[sandboxes[i].Name] > 1 {
```

Comment:
I guess this check could be relaxed by calling promote_subcluster endpoint in the future?

Reply:
Yes, that's correct. We could revisit this. VER-94031 is opened to address this.